inlanecruising_plugin/src/calculation.cpp

```diff
@@ -0,0 +1,308 @@
+#include <ros/ros.h>
+#include <string>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <boost/optional/optional.hpp>
+#include <algorithm>
+#include <tf/transform_datatypes.h>
+#include "inlanecruising_plugin.h"
+#include <tuple>
+#include <cav_msgs/ManeuverPlan.h>
+#include <cav_msgs/LaneFollowingManeuver.h>
+#include <carma_wm/WorldModel.h>
+#include <carma_wm/Geometry.h>
+#include <limits>
+#include <unordered_set>
+
+/*
+#
+# 2d x-axis position of the vehicle center of gravity in meters
+# This position is in a fixed inertial frame which vehicle motion is described in
+#
+float64 X_pos_global
+#
+# 2d y-axis position of the vehicle center of gravity in meters
+# This position is in a fixed inertial frame which vehicle motion is described in
+#
+float64 Y_pos_global
+#
+# The orientation of the vehicle's longitudinal axis in radians
+# This orientation is in a fixed inertial frame which vehicle motion is described in
+#
+float64 orientation
+#
+# longitudinal velocity of the vehicle center of gravity in ms in its body frame
+#
+float64 longitudinal_vel
+
+*/
+
+namespace inlanecruising_plugin
+{
+
+std::vector<PointSpeedPair> points_in_time_boundary(const std::vector<PointSpeedPair>& points, double time_span) {
+  std::vector<PointSpeedPair> output;
+  output.reserve(points.size());
+  double current_time = 0;
+  PointSpeedPair prev_pair;
+  bool first = true;
+  for (const auto& p : points) {
+    if (current_time > time_span) {
+      return output;
+    }
+
+   ROS_WARN_STREAM("Evaluating time of point: " << std::get<0>(p).x() << ", " << std::get<0>(p).y());
+
+
+    if (first) {
+      output.push_back(p);
+      prev_pair = p;
+      first = false;
+      continue;
+    }
+
+    double delta_d = lanelet::geometry::distance2d(std::get<0>(p), std::get<0>(prev_pair));
+
+    current_time += delta_d / std::get<1>(prev_pair);
```

Comment:
Is it valid to assume we're traveling at constant prev_point speed as opposed to average of both point's speeds?